src/append/rolling_file/rotation.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 tison <wander4096@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use time::format_description;
+use time::Duration;
+use time::OffsetDateTime;
+use time::Time;
+
+/// Defines a fixed period for rolling of a log file.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub enum TimeRotation {
```

Comment:
Let's keep the name `Rotation` as we don't introduce other rotation strategy. Also this is a public struct that changing its name would cause downstream software to migrate.

This PR adds tests only, and thus should avoid breaking changes.